src/cairoUtilFuncGen/storage/copyToStorage.ts

```diff
@@ -0,0 +1,231 @@
+import assert from 'assert';
+import {
+  ArrayType,
+  ASTNode,
+  DataLocation,
+  Expression,
+  FunctionStateMutability,
+  generalizeType,
+  getNodeType,
+  StructDefinition,
+  TypeNode,
+  UserDefinedType,
+} from 'solc-typed-ast';
+import { AST } from '../../ast/ast';
+import {
+  CairoFelt,
+  CairoStruct,
+  CairoTuple,
+  CairoType,
+  TypeConversionContext,
+  WarpLocation,
+} from '../../utils/cairoTypeSystem';
+import { TranspilationAbandonedError } from '../../utils/errors';
+import { createCairoFunctionStub, createCallToFunction } from '../../utils/functionGeneration';
+import { mapRange, typeNameFromTypeNode } from '../../utils/utils';
+import { add, CairoFunction, StringIndexedFuncGen } from '../base';
+import { DynArrayGen } from './dynArray';
+
+/*
+  Generates functions to copy data from WARP_STORAGE to WARP_STORAGE
+  The main point of care here is to copy dynamic arrays. Mappings and types containing them
+  cannot be copied from storage to storage, and all types other than dynamic arrays can be
```

Comment:
Is this a current limitation, a limitation which needs to be overcome or a restriction by solidity?

Reply:
Restriction by solidity. I'll update the comment in a future PR